files/external_node_v2.rb

```diff
@@ -342,22 +344,24 @@ def watch_and_send_facts(parallel)
       upload_facts_parallel(http_fact_requests)
     else
       certname = ARGV[0] || raise("Must provide certname as an argument")
-      # send facts to Foreman - enable 'facts' setting to activate
-      # if you use this option below, make sure that you don't send facts to foreman via the rake task or push facts alternatives.
-      #
-      if SETTINGS[:facts]
-        req = generate_fact_request certname, "#{puppetdir}/yaml/facts/#{certname}.yaml"
-        upload_facts(certname, req)
-      end
+
       #
       # query External node
       begin
         result = ""
         Timeout.timeout(tsecs) do
+          # send facts to Foreman - enable 'facts' setting to activate
+          # if you use this option below, make sure that you don't send facts to foreman via the rake task or push facts alternatives.
+          #
+          if SETTINGS[:facts]
+            req = generate_fact_request certname, "#{puppetdir}/yaml/facts/#{certname}.yaml"
+            upload_facts(certname, req)
+          end
+
           result = enc(certname)
           cache(certname, result)
         end
-      rescue TimeoutError, SocketError, Errno::EHOSTUNREACH, Errno::ECONNREFUSED
+      rescue TimeoutError, SocketError, Errno::EHOSTUNREACH, Errno::ECONNREFUSED, FactUploadError
```

Comment:
Could you add some $stderr output to this rescue, or perhaps to the fact upload part? I like the behaviour of using the cache, but there's no indication on running node.rb that it's failing. See 8e86e15.